website/docs/r/budget.html.markdown

```diff
@@ -0,0 +1,127 @@
+---
+layout: "aws"
+page_title: "AWS: aws_budget"
+sidebar_current: "docs-aws-resource-budget"
+description: |-
+  Provides a budget resource.
+---
+
+# aws_budget
+
+Provides a budget resource. Budgets use the cost visualisation provided by Cost Explorer to show you the status of your budgets, to provide forecasts of your estimated costs, and to track your AWS usage, including your free tier usage.
+
+## Example Usage
+
+```hcl
+resource "aws_budget" "ec2" {
+  budget_name           = "budget-ec2-monthly"
+  budget_type           = "spend"
+  limit_amount          = "1200"
+  limit_unit            = "dollars"
+  time_period_end       = "2087-06-15_00:00"
+  time_period_start     = "2017-07-01"
+  time_unit             = "MONTHLY"
+
+  cost_filters {
+    Service = "ec2"
+  }
+}
+```
+
+## Argument Reference
+
+For more detailed documentation about each argument, refer to the [AWS official
+documentation](http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/data-type-budget.html).
+
+The following arguments are supported:
```

Comment:
Each of these arguments should denote `(Required)` or `(Optional)` - someday we'll have automated documentation 🙁